tcms/testruns/static/testruns/js/get.js

```diff
@@ -284,9 +287,14 @@ function renderLink(link) {
     const template = $(linkEntryTemplate.cloneNode(true))
     if (link.is_defect) {
         template.find('.link-icon').addClass('fa fa-bug')
+        const bugTooltip = template.find('.bug-tooltip')
+        bugTooltip.css('visibility', 'visible')
+        fetchBugDetails({ href: link.url }, bugTooltip)
```

Comment:
Missing 3rd parameter which is for page level caching, see usage in `static/js/bugs.js`.

Also the way `renderLink()` calls `fetchBugDetails()` causes all details for all bugs to be loaded during the initial page load while on the TestCase page this is done only when the info icon is clicked. Bug trackers could be slow systems so we'd like to avoid loading data which may not even be shown. 

There are also users who have the bad practice of having between 1500-5000 test cases inside their test plans/test runs while a tester rarely will work with more than 10-20 (they usually split this list between people).

See the usage of `fetchBugDetails()` in bugs.js and try binding it to an on-show/on-click event instead.

Reply:
> Missing 3rd parameter which is for page level caching, see usage in static/js/bugs.js.

see https://github.com/kiwitcms/Kiwi/pull/1873/files#diff-4ff0c8d0df45ac317c8f001b07ae4f11R51

I made the third parameter to default to `bugDetailsCache` in `bugs.js`. This way the consumer should not pass it every time and different pages can share the cache.

Reply:
> This way the consumer should not pass it every time and different pages can share the cache.

I'm not sure if this will work. If you have 2 pages loaded in parallel they will both load the bugs.js file and create a local (to them) bugsCache variable. So doesn't look like there will be any cache sharing. 

Note: I didn't see the default parameter but I guess the remaining invocations of this function should be adjusted to make use of it.

Reply:
Note: The previous comment about cache doesn't address the fact that bug details are loaded unconditionally for every bug listed on the page regardless of the fact if the user will ever click on the info icon.

Reply:
I know, I adressed that - https://github.com/kiwitcms/Kiwi/pull/1873/files#diff-e35e7234b4311cd0d2d5fa2bfef17fb1R309-R311